### ISSUE TITLE
replication failed: pull manifest returns 404

### ISSUE THREAD

**Issue by ArthurChiao** [goharbor/harbor#5166]
jobservice sync image failed due to **pull manifest from local registry failed**, detailed registry error: `404 tag unknown, tag=xxx`.

Where the error is raised:
https://github.com/vmware/harbor/blob/master/src/jobservice/job/impl/replication/transfer.go#L228

Root cause: https://github.com/docker/distribution/issues/2625

**Comment by moooofly** [comment]
good job!

**Comment by steven-zou** [comment]
@ywk253100 Please take a look at the root cause and its related influences and confirm if we can do something to fix or improve that part. Thanks

**Comment by ArthurChiao** [comment]
According to my test, retry again after a short time solves the problem.

psudo code:

```

retry := 0
interval := 50 * time.Millisecond

for retry < 5 {
    digest, mediaType, payload, err := pullManifest()
    if err == nil {
        break
    }
    
    if strings.Contain(err.Error(), "404") {
            time.Sleep(interval)
        
            retry++
            interval *= 2
            continue
    } else {
        log.Errorf("pull manifest failed: %s", err)
        break
    }
}

if retry >= 5 {
    log.Errorf("pull manifest failed: max retry reached")
}

```

**Comment by reasonerjt** [comment]
Fixed in #6223
